### ISSUE TITLE
Error when saving audit entity that has new instance of inherited property

### ISSUE THREAD

**Issue by mrlife** [thepirat000/Audit.NET#384]
**Describe the bug**
When inheriting from a base class and creating a new instance of an inherited property, e.g. `Id`, the following error is thrown.

> "An item with the same key has already been added. Key: Id"

Here is example code. Usually, `Id` should be `int`, however, in this case, it needs to be `string`, so I create a `new` `Id`.

I would guess the issue occurs during the property assignments in `Map()` (e.g. assigning values from `Post` to `Audit_Post`), and it is trying to use the base `int Id` as well as the `string Id`, but I'm not 100%. I can't seem to figure out a good way to debug this.

```c#
// base class
public abstract class BaseAudit
{
    public int Id { get; set; }
    public string AuditAction { get; set; }
}

// audit model
public class Audit_Post : BaseAudit
{
    public new string Id { get; set; }
    public string Action { get; set; } // Insert, Delete or Update
    public int PostId { get; set; }
    public string Name { get; set; }
}
```

**To Reproduce**
This error occurs during the call to `SaveChanges()` after the regular entity, e.g. `Post`, has successfully saved.

I get the error whether I use `Map()`, `MapTable()`, or `MapExplicit()` for `Post` and `Audit_Post`.

**Expected behavior**
The `int Id` is ignored and the `string Id` is used by the library when assigning values to the audit entity.

**Libraries (specify the Audit.NET extensions being used including version):**
 - Audit.EntityFramework: 16.5.5

**Target .NET framework:**
 - .NET 5

**Comment by thepirat000** [comment]
Yes, reflection will get both the hidden and the new property, and since there is a `.ToDictionary(prop => prop.Name)`, it fails.

I think I can fix that by ignoring the hidden base properties on the property matching. 

But note that the property matching will fail if your audited entity has an incompatible property type i.e. 
```#
public class Post 
{ 
    public int Id { get; set; }
}
```

**Comment by mrlife** [comment]
This looks great. I'll try it out as soon as it's available. Thank you!

**Comment by thepirat000** [comment]
Fixed on version 16.5.6, please upgrade your references and re-test

**Comment by mrlife** [comment]
This works great. Thank you!
